lerobot/common/policies/vla/modeling_vla.py

```diff
@@ -109,53 +111,94 @@ def forward(self, batch: dict[str, torch.Tensor]) -> dict[str, torch.Tensor]:
 
 
 class VLA(nn.Module):
-    def __init__(self, config: VLAConfig):
+    def __init__(self, config: VLAConfig, precision: torch.dtype = torch.float32):
         super().__init__()
         self.chunk_size = config.chunk_size
         self.action_decoder_name = config.action_decoder.get("name", "act")
-        if self.action_decoder_name == "act":
-            action_decoder_config = OmegaConf.create(config.action_decoder)
-            self.action_decoder = ACTDecoder(action_decoder_config)
-            self.decoder_pos_embed = nn.Embedding(config.chunk_size, config.action_decoder["dim_model"])
+        self.use_robot_state = "observation.state" in config.input_shapes
+        if "act" in self.action_decoder_name:
```

Comment:
Is this related to the new ActEncoderDecoder? 

To me this **if-else block related to act** looks a bit confusing, we could rather rename it completely or structure in this way




 

``` 
if "act" in self.action_decoder_name:
   action_decoder_config = OmegaConf.create(config.action_decoder)       
   if self.action_decoder_name == "act_decoder":
          # Use standalone ACTDecoder
          self.action_decoder = ACTDecoder(action_decoder_config)
          self.decoder_pos_embed = nn.Embedding(config.chunk_size, config.action_decoder["dim_model"])
      else:
          # Use ACTEncoderDecoder, decide whether to include the encoder
          use_encoder = "decoder" not in self.action_decoder_name
          self.action_decoder = ACTEncoderDecoder(action_decoder_config, use_encoder=use_encoder)
```
          
Or even better: if ActDecoder is equivalent to `ActEncoderDecoder `with `use_encoder=False` then we can remove the use of ActDecoder completely and leave only the option use_encoder for act to avoid confusion and redundancy.

Reply:
Yes, ACTEncoderDecoder is more general but I kept it to be able your old checkpoints without changing its keys. If this is not needed anymore (e.g. we trained a good VLA with this branch) we should keep only the ACTEncoderDecoder

Reply:
Yes, ACTEncoderDecoder is more general but I kept it to be able your old checkpoints without changing its keys. If this is not needed anymore (e.g. we trained a good VLA with this branch) we should keep only the ACTEncoderDecoder